contrib/scripts/consolidate_go_stacktrace.py

```diff
@@ -59,6 +62,12 @@ def get_hashable_stack_value(stack):
         metavar='PATH',
         nargs='?',
         help='Read and parse this file. Specify \'-\' or omit this option for stdin.')
+    parser.add_argument(
+        '-s',
+        '--source-dir',
+        nargs=1,
+        default=cilium_source,
+        help='Rewrite Cilium source paths to refer to this directory')
```

Comment:
Call out that it's an absolute path?

Reply:
I don't know if it needs to be an absolute path? There's nothing particularly special here, it just does a blind path substitution for whatever path you specify. My intended usage is you just use `-s $PWD` to specify the current directory when you're in the cilium directory and it will update the paths to that one. I think you could just use a relative path, but then I don't know if your editor will be smart enough to figure out those paths if you ctrl+click.

Reply:
^^ I just tried it out with vscode and it works with a relative path too. This is actually a little quicker, just `consolidate_go_stacktrace.py my_stacktrace -s .` and it works.

```
$ contrib/scripts/consolidate_go_stacktrace.py ipcache-policy-deadlock.txt -s . | head -n 20
114 unique stack traces
18 occurences. Sample stack trace:
sync.runtime_SemacquireMutex(0x30?, 0x0?, 0xc0033a3620?)
        /usr/local/go/src/runtime/sema.go:77 +0x25
sync.(*RWMutex).RLock(...)
        /usr/local/go/src/sync/rwmutex.go:71
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).unconditionalRLock(...)
        ./pkg/endpoint/lock.go:53
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).HasLabels(0xc000ee2000?, 0x2807b69?)
        ./pkg/endpoint/endpoint.go:1028 +0x59
github.com/cilium/cilium/daemon/cmd.(*Daemon).getEndpointList.func1(0xc0006f9e60?, 0xc0030ff590?, 0xc0030ff5a0?)
        ./daemon/cmd/endpoint.go:96 +0x71
created by github.com/cilium/cilium/daemon/cmd.(*Daemon).getEndpointList
        ./daemon/cmd/endpoint.go:94 +0x16f
...
```

Reply:
Even better!